lib/ui/src/components/preview/tools/menu.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { IconButton, Icons, Separator } from '@storybook/components';
+import { Consumer, Combo } from '@storybook/api';
+import { Addon } from '@storybook/addons';
+
+const menuMapper = ({ api, state }: Combo) => ({
+  isVisible: state.layout.showNav,
+  toggle: api.toggleNav,
+});
+
+export const menuTool: Addon = {
+  title: 'menu',
+  id: 'menu',
+  match: ({ viewMode }) => viewMode === 'story',
+  render: () => (
+    <>
+      <Consumer filter={menuMapper}>
+        {({ isVisible, toggle }) =>
+          isVisible && (
```

Comment:
I think since `isVisible` is relating to the side nav, we want to display the button when the sidebar is not visible. Currently, the button disappears when I close the sidebar.
```suggestion
          !isVisible && (
```

Reply:
Sorry about that! It broke after the previous commit on the conditional rendering.